spec/factories.rb

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FactoryBot.define do
+  # Using YAML parsing short circuits the object lifecycle.
+  # We don't have proper initialize methods, because our
+  # objects just appear post-YAML parsing with
+  # all of the correct values.
+  #
+  # FactoryBot should create its test objects in the
+  # same manner.
+  initialize_with do
+    obj = new
+    attributes.each { |k, v| obj.instance_variable_set("@#{k.id2name}", v) }
+    # TODO(alexstephen): Build out a better default factory that can successfully
```

Comment:
What would make a factory a default factory instead of a non-default factory?

Reply:
All of the values in {} are the default values. So, I wrote down that it was the default factory, even though not really a thing. Please ignore that weird wording on my part.

Reply:
Second question: Why are they defaults if they are values we type in every time?

Reply:
We aren't typing them in everytime? Not following your question